wasp/apps/template.py

```diff
@@ -44,23 +44,30 @@ def __init__(self):
         pass
 
     def foreground(self):
-        """Activate the application."""
+        """Called when the application is launched, for example by pressing
+        its icon in the launcher."""
         self._draw()
         wasp.system.request_event(wasp.EventMask.TOUCH |
                                   wasp.EventMask.SWIPE_UPDOWN |
                                   wasp.EventMask.BUTTON)
         wasp.system.request_tick(1000)
 
     def background(self):
-        """De-activate the application."""
+        """Called when the leaving the application, either because of screen
+        timeout of directly by the user."""
         pass
 
     def sleep(self):
-        """Notify the application the device is about to sleep."""
+        """Called by the system to notify the application that the screen is
+        about to be turned off. Most apps should not implement it.
+        If not implemented or returning True: goes back to the default clock
+        application. If implemented and returning False: turning the screen
+        back on will show the current application"""
         return False
 
     def wake(self):
-        """Notify the application the device is waking up."""
+        """Notify the application that the screen is waking up. Generally used
+        to redraw the screen or to re-enable ticks."""
```

Comment:
I thought the tick was re-enabled automatically by the system.

Reply:
I removed the section about tics then